### PR TITLE
Mix - Deploydialog

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

**Comment by yann300** [comment]
fix for #2441 #2442

File: mix/ClientModel.cpp [hunk 1]
```diff
@@ -295,36 +302,40 @@ void ClientModel::finalizeBlock()
 	}
 }
 
+TransactionSettings ClientModel::transaction(QVariant _tr)
```
Comment (arkpar) [inline mix/ClientModel.cpp]: QVariant const&

File: mix/ClientModel.h [hunk 2]
```diff
@@ -266,6 +270,7 @@ private slots:
 	void finalizeBlock();
 	void stopExecution();
 	void setupExecutionChain();
+	TransactionSettings transaction(QVariant _tr);
```
Comment (arkpar) [inline mix/ClientModel.h]: function should be made const if possible

File: mix/MixClient.cpp [hunk 3]
```diff
@@ -124,22 +123,16 @@ Transaction MixClient::replaceGas(Transaction const& _t, u256 const& _gas, Secre
 	return ret;
 }
 
-void MixClient::executeTransaction(Transaction const& _t, State& _state, bool _call, bool _gasAuto, Secret const& _secret)
+ExecutionResult MixClient::debugTransaction(Transaction const& _t, State const& _state, LastHashes _lastHashes, bool _call)
```
Comment (arkpar) [inline mix/MixClient.cpp]: LastHashes const&

File: mix/qml/BlockChain.qml [hunk 4]
```diff
@@ -283,6 +314,43 @@ ColumnLayout {
...
+						rebuild.startBlinking()
+						blinkReasons.push(reason)
+					}
+
+					function containsRebuildCause(reason)
+					{
+						for (var c in blinkReasons)
+						{
+							if (blinkReasons[c] === reason)
+							{
```
Comment (arkpar) [inline mix/qml/BlockChain.qml]: Extra brackets here and in few places below

**Pull request merged**